packages/core/android/src/main/java/com/datadog/reactnative/DdSdk.kt

```diff
@@ -60,8 +88,164 @@ class DdSdk(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(
      */
     @ReactMethod
     fun setTrackingConsent(trackingConsent: String, promise: Promise) {
-        nativeInstance.setTrackingConsent(trackingConsent)
+        datadog.setTrackingConsent(buildTrackingConsent(trackingConsent))
         promise.resolve(null)
     }
 
+    // endregion
+
+    // region Internal
+
+    private fun configureSdkVerbosity(configuration: DdSdkConfiguration) {
+        val verbosityConfig = configuration.additionalConfig?.get(DD_SDK_VERBOSITY) as? String
+        val verbosity = when (verbosityConfig?.lowercase(Locale.US)) {
+            "debug" -> Log.DEBUG
+            "info" -> Log.INFO
+            "warn" -> Log.WARN
+            "error" -> Log.ERROR
+            else -> null
+        }
+        if (verbosity != null) {
+            datadog.setVerbosity(verbosity)
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    private fun buildConfiguration(configuration: DdSdkConfiguration): Configuration {
```

Comment:
I think you could extract some methods here, this is too big.

Reply:
Good point. I think this can be done in another PR as it does not have a big impact yet, and without proper tests in place I have more risks to introduce a regression.

Reply:
yes, better to do it in dedicated PR. Anyway this is how it is in the bridge now.